std/concurrency.d

```diff
@@ -2268,6 +2273,26 @@ private
 
 version (unittest)
 {
+    void changeScheduler(Scheduler s)
+    {
+        import core.memory : GC;
+        import core.thread : Thread;
+
+        bool sleepFirst = false;
+        while (Thread.getAll().length != 1)
+        {
+            // wait for all other threads to terminate, use GC.collect to help progress threads by
+            // running finalizers (which might trigger OwnerTerminated or LinkTerminated that other
+            // threads are waiting for)
+            if (sleepFirst)
+                Thread.sleep(dur!("msecs")( 10 ));
+            else
+                sleepFirst = true;
+            GC.collect;
```

Comment:
> Do we really only send Owner/LinkTerminated messages when the thread object get's collected?

They get sent when the module destructor is run for threads, and via ```scope(exit)``` for fibers, so the comment I added must be wrong.

I have look at the code again (it's been so long since I made this PR) to see whether this was just a hack for force bad tests to hang (instead of random failures,) or whether it was necessary.